run_project_tests.py

```diff
@@ -106,10 +154,12 @@ def display_name(self) -> str:
             return '{}   ({})'.format(self.path.as_posix(), self.name)
         return self.path.as_posix()
 
-    def __lt__(self, other: T.Any) -> T.Union[bool, type(NotImplemented)]:
+    def __lt__(self, other: T.Any) -> bool:
```

Comment:
This change is wrong, all of the rich comparison methods return a Untion of bool and NotImplemented

Reply:
This is the only way to make both mypy and python 3.5.2 happy. See https://github.com/mesonbuild/meson/pull/6438#issuecomment-584644255

Reply:
<rant>I'm seriously tired of bending over backwards to try to make ancient versions of ubuntu work with modern code bases </rant>.

I've seen mypy complain that bool is not right in that case, maybe they've changed it or an old version doesn't catch it?

Reply:
I am running mypy 0.761 and Github actions also passes. Alternatively, we could get rid of all python 3.5 issues with #6438 :)

Reply:
For now I guess I'll drop my complaint. I would really like to have that patch and be able to take advantage of 3.6 improvements. There's a lot of them :)